File: MomentumiOS/MomentumiOSTests/payments/FakePaymentController.swift
//
//  FakePaymentController.swift
//  MomentumiOSTests
//
//  Created by Mwai Banda on 8/1/22.
//  Copyright © 2022 Momentum. All rights reserved.
//

@testable import MomentumSDK

class FakePaymentController: PaymentController {
    func checkout(request: PaymentRequest, onCompletion: @escaping (PaymentResponse) -> Void) {
        let response: PaymentResponse = .init(
            publishableKey: "100101-111",
            paymentIntent: "Intent",
            customer: "Mwai Banda",
            ephemeralKey: "0007722"
        )
        onCompletion(response)
    }
}

File: MomentumAndroid/src/test/java/com/mwaibanda/momentum/android/presentation/payment/FakePaymentController.kt
package com.mwaibanda.momentum.android.presentation.payment

import com.mwaibanda.momentum.domain.controller.PaymentController
import com.mwaibanda.momentum.domain.models.Payment
import com.mwaibanda.momentum.domain.models.PaymentResponse
import com.mwaibanda.momentum.utils.DataResponse

class FakePaymentController: PaymentController {
    override fun checkout(request: Payment, onCompletion: (DataResponse<PaymentResponse>) -> Unit) {
        onCompletion(DataResponse.Success(PaymentResponse(
            publishableKey = "1001-001",
            paymentIntent = "Intent",
            customer = request.fullname,
            ephemeralKey = "10001"
        )))
    }
}